src/slidedown/ChannelCaptureContainer.ts

```diff
@@ -137,30 +132,90 @@ export default class ChannelCaptureContainer {
     }
 
     private initializePhoneInputLibrary(): void {
-        const onesignalSmsInput = document.querySelector("#iti-onesignal-sms-input");
+        const onesignalSmsInput = getDomElementOrStub(`#${CHANNEL_CAPTURE_CONTAINER_CSS_IDS.onesignalSmsInput}`);
         if (onesignalSmsInput && !!window.intlTelInput) {
             this.itiOneSignal = window.intlTelInput(onesignalSmsInput, {
                 autoPlaceholder: "off",
                 separateDialCode: true
             });
-            onesignalSmsInput.addEventListener('keyup', () => {
-                this.smsInputFieldIsValid = this.itiOneSignal.isValidNumber() ||
-                    (<HTMLInputElement>onesignalSmsInput).value === "";
-            });
-            // covers case where number is typed, then country is changed
-            onesignalSmsInput.addEventListener('blur', () => {
-                this.smsInputFieldIsValid = this.itiOneSignal.isValidNumber() ||
-                    (<HTMLInputElement>onesignalSmsInput).value === "";
-            });
         } else {
             Log.error("OneSignal: there was a problem initializing International Telephone Input");
         }
     }
 
-    // static functions
+    private addSmsInputEventListeners(): void {
+        const smsInput = getDomElementOrStub(`#${CHANNEL_CAPTURE_CONTAINER_CSS_IDS.onesignalSmsInput}`);
 
+        smsInput.addEventListener('keyup', () => {
+            this.smsInputFieldIsValid = this.itiOneSignal.isValidNumber() ||
+                (<HTMLInputElement>smsInput).value === "";
+
+            this.updateValidationOnInputChange(DelayedPromptType.Sms);
+        });
+
+        // handles case where number is typed, then country is changed after
+        smsInput.addEventListener('blur', () => {
+            this.smsInputFieldIsValid = this.itiOneSignal.isValidNumber() ||
+                (<HTMLInputElement>smsInput).value === "";
+
+            this.updateValidationOnInputChange(DelayedPromptType.Sms);
+        });
+    }
+
+    private addEmailInputEventListeners(): void {
+        const emailInput = getDomElementOrStub(`#${CHANNEL_CAPTURE_CONTAINER_CSS_IDS.onesignalEmailInput}`);
+
+        // TO DO: refactor all this
+        emailInput.addEventListener('keyup', () => {
+            const emailValue = (<HTMLInputElement>emailInput).value;
+            this.emailInputFieldIsValid = ChannelCaptureContainer.validateEmailInputWithReturnVal(emailValue);
+
+            this.updateValidationOnInputChange(DelayedPromptType.Email);
+        });
+    }
+
+    private updateValidationOnInputChange(type: DelayedPromptType): void {
```

Comment:
Should be two methods. `updateValidationOnEmailChange`  and `updateValidationOnSMSChange`. After that we could clean up more by then making a shared function these to can use to update the CSS ect.